### PR TITLE
Refactor Withdrawal Extrinsic 

### PR DESCRIPTION
**Pull request by felixfaisal**
This PR refactors withdrawal extrinsic wherein it utilises the BTreeMap provided in the snapshot rather than a Bounded. This PR also includes a helper function that returns the vector of withdrawals for a user.

### CODE REVIEW

**Comment by serhii-temchenko** [comment]
I assume this PR is depends on [this](https://github.com/Polkadex-Substrate/polkadex-primitives/pull/23) PR and it should be discussed as commented [here](https://github.com/Polkadex-Substrate/polkadex-primitives/pull/23#issuecomment-1205690504) and I assume after merge should be bumped version (tag) of this dependency.

File: pallets/ocex/src/lib.rs [hunk 1]
```diff
@@ -380,10 +383,12 @@ pub mod pallet {
 				signature.verify(bytes.as_slice(), &enclave),
 				Error::<T>::EnclaveSignatureVerificationFailed
 			);
-			<Withdrawals<T>>::insert(snapshot.snapshot_number, snapshot.withdrawals);
+			let mut withdrawal_map: BoundedBTreeMap<T::AccountId, BoundedVec<Withdrawal<T::AccountId, BalanceOf<T>>, WithdrawalLimit>, WithdrawalLimit> = BoundedBTreeMap::try_from(snapshot.withdrawals.clone()).expect("Bounded BTreeMap could not be created");
```
Comment (serhii-temchenko) [inline pallets/ocex/src/lib.rs]: Are you sure that we need to panic here and not returning error?

**Review by serhii-temchenko — DISMISSED** [review]

File: pallets/ocex/src/lib.rs (hunk 1 above)
Comment (Gauthamastro) [reply pallets/ocex/src/lib.rs]: @felixfaisal there should be no possibility for a panic, even by a unhandled arithmetic operation is allowed in runtime code.

**Review by Gauthamastro — CHANGES_REQUESTED** [review]
Panics must be removed

File: pallets/ocex/src/lib.rs [hunk 2]
```diff
@@ -383,12 +383,16 @@ pub mod pallet {
...
 				Error::<T>::EnclaveSignatureVerificationFailed
 			);
-			let mut withdrawal_map: BoundedBTreeMap<T::AccountId, BoundedVec<Withdrawal<T::AccountId, BalanceOf<T>>, WithdrawalLimit>, WithdrawalLimit> = BoundedBTreeMap::try_from(snapshot.withdrawals.clone()).expect("Bounded BTreeMap could not be created");
+			let mut withdrawal_map: BoundedBTreeMap<T::AccountId, BoundedVec<Withdrawal<T::AccountId, BalanceOf<T>>, WithdrawalLimit>, WithdrawalLimit> = BoundedBTreeMap::new();
+			if let Ok(value) = BoundedBTreeMap::<T::AccountId, BoundedVec<Withdrawal<T::AccountId, BalanceOf<T>>, WithdrawalLimit>, WithdrawalLimit>::try_from(snapshot.withdrawals.clone()){
+				withdrawal_map = value;
+			}
+			else {
+				return Err(Error::<T>::StorageOverflow.into());
+			}
```
Comment (serhii-temchenko) [inline pallets/ocex/src/lib.rs]: Can this be simplified like:
```rust 
let mut withdrawal_map: BoundedBTreeMap<T::AccountId, BoundedVec<Withdrawal<T::AccountId, BalanceOf<T>>, WithdrawalLimit>, WithdrawalLimit> = 
    BoundedBTreeMap::try_from(snapshot.withdrawals.clone()).ok_or(Error::<T>::StorageOverflow.into())?;

File: pallets/ocex/src/lib.rs (hunk 1 above)
Comment (felixfaisal) [reply pallets/ocex/src/lib.rs]: Error Handling done ✅

File: pallets/ocex/src/lib.rs (hunk 2 above)
Comment (felixfaisal) [reply pallets/ocex/src/lib.rs]: Let me try this

Comment (felixfaisal) [reply pallets/ocex/src/lib.rs]: I'm getting errors while trying to do this

Comment (felixfaisal) [reply pallets/ocex/src/lib.rs]: @serhii-temchenko So `ok_or` here is returning the `Error::<T>::StorageOverflow.into()` and storing into withdrawal_map, when we have defined the type to be BoundedBTreeMap and it cannot store the error type, so this code will not compile.

Comment (serhii-temchenko) [reply pallets/ocex/src/lib.rs]: How you are compiling this? I have issue with build conflicting version of `syn` crate in this branch even when referring your branch in primitives.

Comment (felixfaisal) [reply pallets/ocex/src/lib.rs]: So I've not updated the Cargo.toml with new dependencies and that is why it is not compiling for you

Comment (felixfaisal) [reply pallets/ocex/src/lib.rs]: I will do that as soon as I update the upstream branch of Polkadex primitives

Comment (felixfaisal) [reply pallets/ocex/src/lib.rs]: This is done ✅

File: pallets/ocex/rpc/Cargo.toml [hunk 3]
```diff
@@ -16,5 +16,5 @@ sp-blockchain = {   git = "https://github.com/paritytech/substrate", branch = "p
 sp-api = {   git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.19", default-features = false }
 sp-runtime = {   git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.19", default-features = false }
 pallet-ocex-primitives = {path = "../common", default-features = false}
-pallet-ocex-lmp-runtime-api = {path = "runtime-api" }
-polkadex-primitives = { git = "https://github.com/Polkadex-Substrate/polkadex-primitives.git",  branch="fx-polkadot-v0.9.19" , default-features = false }
+pallet-ocex-lmp-runtime-api = {path = "runtime-api", default-features = false }
+polkadex-primitives = { git = "https://github.com/Polkadex-Substrate/polkadex-primitives.git",  branch="fx-polkadot-v0.9.19" , default-features = false }
```
Comment (serhii-temchenko) [inline pallets/ocex/rpc/Cargo.toml]: Empty trailing line.

File: node/src/main.rs [hunk 4]
```diff
@@ -8,6 +8,7 @@ mod cli;
 mod command;
 mod rpc;
 
+
```
Comment (serhii-temchenko) [inline node/src/main.rs:11]: Please remove.

File: node/Cargo.toml [hunk 5]
```diff
@@ -84,7 +85,7 @@ sc-transaction-pool-api = { git = "https://github.com/paritytech/substrate", bra
 sp-authority-discovery = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.19" }
 
 #Primitives
-polkadex-primitives = { git = "https://github.com/Polkadex-Substrate/polkadex-primitives.git", branch = 'polkadot-v0.9.19'}
+polkadex-primitives = { git = "https://github.com/Polkadex-Substrate/polkadex-primitives.git", branch = "fx-polkadot-v0.9.19", default-features = false }
```
Comment (serhii-temchenko) [inline node/Cargo.toml]: Why this is referring this branch? I think you need to change to  `polkadot-v0.9.19`

**Review by serhii-temchenko — CHANGES_REQUESTED** [review]

**Comment by pavankanteti** [comment]
@felixfaisal to add a comment and close this

**Comment by pavankanteti** [comment]
will be resolve in #479

**Pull request closed without merging**